lib/src/room.dart

```diff
@@ -310,25 +317,63 @@ class KurentoRoom extends ChangeNotifier implements Room {
     // Eventually, we will create a separate connection for screen sharing. That requires changes to Platform and Dash,
     // so for now, we start presenting by replacing the Explorer's video track with the display video track.
     _presentationStream = await displayStream.clone();
+
+    // Since we don't show the localStream's video and the presentation simultaneously, disabled it to free up some CPU.
+    _localStream!.getVideoTracks()[0].enabled = false;
     if ((_localStream?.getAudioTracks().isNotEmpty ?? false) && _presentationStream!.getAudioTracks().isEmpty) {
-      await _presentationStream?.addTrack(_localStream!.getAudioTracks()[0]);
+      // Set the presentation audio track to keep audio.
+      await _presentationStream!.addTrack(_localStream!.getAudioTracks()[0]);
     }
-    MediaStreamTrack presentationTrack = displayStream.getVideoTracks()[0];
-    await _connectionByTrackId[_localTrackId]!.replaceVideoTrack(_isPresentationMuted ? null : presentationTrack);
+
+    if (kIsWeb) {
+      MediaStreamTrack presentationTrack = _presentationStream!.getVideoTracks()[0];
+      await _connectionByTrackId[_localTrackId]!.replaceVideoTrack(_isPresentationMuted ? null : presentationTrack);
+    } else {
+      // Hack! Since the "replaceVideoTrack" function causes sporadically a pixelation of the video on mobile, we are
+      // recreating the video track from scratch.
+      await _client.deleteTrack(_serviceRequest.roomId, _serviceRequest.participantId, _localTrackId!);
+      await _createOutgoingTrack();
+    }
+
     await _updateParticipantStatus();
-    _log.info('started presenting track=${presentationTrack.label}');
+    _log.info('started presenting track=${_presentationStream!.getVideoTracks()[0].label}');
   }
 
   @override
   Future<void> stopPresenting() async {
     if (_localTrackId == null) {
       return;
     }
-    // Until we create a separate connection for screen sharing, we stop presenting by restoring the Explorer's video
-    // track.
-    await _connectionByTrackId[_localTrackId]!
-        .replaceVideoTrack(_isVideoMuted || !_hasVideoTrack ? null : _localStream!.getVideoTracks()[0]);
-    _presentationStream = null;
+
+    _localStream!.getVideoTracks()[0].enabled = true;
+    if (null != _presentationStream) {
+      // Properly disposing of the presentation tracks.
+      if (_presentationStream!.getAudioTracks().isNotEmpty) {
+        if (null != _localStream &&
+            _presentationStream!.getAudioTracks()[0].id == _localStream!.getAudioTracks()[0].id) {
+          // If we are using _localStream's audio track, don't destroy it!
+          await _presentationStream!.removeTrack(_localStream!.getAudioTracks()[0]);
+        } else {
+          await _presentationStream!.getAudioTracks()[0].stop();
+        }
+      }
+      await _presentationStream!.getVideoTracks()[0].stop();
+      await _presentationStream!.dispose();
+      _presentationStream = null;
+    }
```

Comment:
`_presentationStream` should not contain `_localStream`'audio at this point: remove it!